src/components/ListItemTagInput.js

```diff
@@ -1,41 +1,77 @@
-import React from 'react'
+import React, { Component, PropTypes } from 'react'
 import { get } from 'lodash'
 import { connect } from 'react-redux'
 import TagInput from './TagInput'
 import { typeahead } from '../actions'
 import { trackEvent, ADDED_SKILL } from '../util/analytics'
-const { func, object, string } = React.PropTypes
-
-const ListItemTagInput = connect(
-  ({ typeaheadMatches }, { type }) => ({matches: get(typeaheadMatches, type)})
-)(({ dispatch, matches, type, person, update, filter, className, context }) => {
-  let list = person[type] || []
-  let tags = list.map(x => ({
-    id: x,
-    name: x,
-    label: type === 'tags' ? '#' + x : x
-  }))
-  let add = item => {
+
+export class ListItemTagInput extends Component {
+  static propTypes = {
+    type: PropTypes.string.isRequired,
+    person: PropTypes.object.isRequired,
+    update: PropTypes.func.isRequired,
+    dispatch: PropTypes.func.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      list: props.person[props.type] || []
```

Comment:
it's not clear to me that local state is required at all here -- don't your local changes to `this.state.list` exactly mirror changes you're making to the app state?